libraries/AP_InertialSensor/AP_InertialSensor_Invensensev3.cpp

```diff
@@ -1074,10 +1074,28 @@ bool AP_InertialSensor_Invensensev3::hardware_init(void)
         // disable STC
         uint8_t reg = register_read_bank_icm456xy(INV3BANK_456_IPREG_TOP1_ADDR, 0x68);  // I3C_STC_MODE b2
         register_write_bank_icm456xy(INV3BANK_456_IPREG_TOP1_ADDR, 0x68, reg & ~0x04);
-    } else if (inv3_type == Invensensev3_Type::ICM42688) {
-        // fix for the "stuck gyro" issue
-        const uint8_t v = register_read(INV3REG_42688_INTF_CONFIG1);
-        register_write(INV3REG_42688_INTF_CONFIG1, (v & 0x3F) | 0x40);
+    }
+
```

Comment:
I think where this is potentially violates 12.9 which applies to all these sensors - "The only register settings that user can modify during sensor operation are for ODR selection, FSR selection, and sensor mode changes (register parameters GYRO_ODR, ACCEL_ODR, GYRO_FS_SEL, ACCEL_FS_SEL, GYRO_MODE, ACCEL_MODE)". It does not apply to the other sensors making changes here. Either you should explicitly disable accel/gyros first or it might be easier to move this into set_filter_and_scaling() to avoid this,

Reply:
done